core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1536,29 +1537,38 @@ class ReplicaManager(val config: KafkaConfig,
     }
   }
 
-  def electPreferredLeaders(controller: KafkaController,
-                            partitions: Set[TopicPartition],
-                            responseCallback: Map[TopicPartition, ApiError] => Unit,
-                            requestTimeout: Long): Unit = {
+  def electLeaders(
+    controller: KafkaController,
+    partitions: Set[TopicPartition],
+    electionType: ElectionType,
+    responseCallback: Map[TopicPartition, ApiError] => Unit,
+    requestTimeout: Long
```

Comment:
The type here is `Long`, but the request actually specifies `Int`. Not sure it matters too much.

Another issue for a follow-up, but we take this value as is from the client without validation. This is probably a common pattern for admin APIs. Possibly we could enforce a range of values. At a minimum, it would be nice to take negative values out of the picture.